### DISCUSSION TITLE
Accessing route query params on the initial page load

### DISCUSSION THREAD

**Discussion by vmihailenco** [vuepress/core#990] [Q&A]
It looks like query params are dropped on the initial page load, for example, https://vuepress2.netlify.app/guide/getting-started.html?foo=bar is redirected to https://vuepress2.netlify.app/guide/getting-started.html and `route.query` is empty in `setup` function.

It only happens on the initial page load. Subsequently, `route.query` can be accessed without an issue.

Are there any workarounds? I am using VuePress 2.0.0-beta.49 with Vite loader.

**Comment by Mister-Hope** [comment]
Is the issue still exisit now? If now just mark this as answer, if so open a new issue. Probably related to header active plugin
